### PR TITLE
Update of ItemManager

### PR DESCRIPTION
**Pull request by LogGits**
Updated:
- wood pressureplates
- wood buttons
- carved pumpkin
and a few others

### CODE REVIEW

File: src/main/java/com/gamingmesh/jobs/CMILib/ItemManager.java [hunk 1]
```diff
@@ -611,14 +611,14 @@ public static String getRealNameByType(EntityType type) {
...
-	ACACIA_BUTTON(-1, -1, 13993, "Acacia Button", ""),
+	ACACIA_BUTTON(143, 0, 13993, "Acacia Button", "wooden_button"),
 	ACACIA_DOOR(430, 0, 23797, "Acacia Door", "ACACIA_DOOR_ITEM"),
 	ACACIA_FENCE(192, 0, 4569, "Acacia Fence", ""),
 	ACACIA_FENCE_GATE(187, 0, 14145, "Acacia Fence Gate", ""),
 	ACACIA_LEAVES(161, 0, 16606, "Acacia Leaves", "LEAVES_2"),
 	ACACIA_LOG(162, 0, 8385, "Acacia Log", "LOG_2"),
 	ACACIA_PLANKS(5, 4, 31312, "Acacia Wood Plank", "Acacia Planks"),
-	ACACIA_PRESSURE_PLATE(-1, -1, 17586, "Acacia Pressure Plate", ""),
+	ACACIA_PRESSURE_PLATE(72, 0, 17586, "Acacia Pressure Plate", "Wooden_Presure_Plate"),
```
Comment (montlikadani) [inline src/main/java/com/gamingmesh/jobs/CMILib/ItemManager.java]: `Presure` need double `s`.

File: src/main/java/com/gamingmesh/jobs/CMILib/ItemManager.java [hunk 2]
```diff
@@ -646,14 +646,14 @@ public static String getRealNameByType(EntityType type) {
...
-	BIRCH_BUTTON(-1, -1, 26934, "Birch Button", ""),
+	BIRCH_BUTTON(143, 0, 26934, "Birch Button", "wooden_button"),
 	BIRCH_DOOR(428, 0, 14759, "Birch Door", "BIRCH_DOOR_ITEM"),
 	BIRCH_FENCE(189, 0, 17347, "Birch Fence", ""),
 	BIRCH_FENCE_GATE(184, 0, 6322, "Birch Fence Gate", ""),
 	BIRCH_LEAVES(18, 2, 12601, "Birch Leaves", "LEAVES"),
 	BIRCH_LOG(17, 2, 26727, "Birch Log", "LOG"),
 	BIRCH_PLANKS(5, 2, 29322, "Birch Wood Plank", "Birch Planks"),
-	BIRCH_PRESSURE_PLATE(-1, -1, 9664, "Birch Pressure Plate", ""),
+	BIRCH_PRESSURE_PLATE(72, 0, 9664, "Birch Pressure Plate", "Wooden_Presure_Plate"),
```
Comment (montlikadani) [inline src/main/java/com/gamingmesh/jobs/CMILib/ItemManager.java]: Same

File: src/main/java/com/gamingmesh/jobs/CMILib/ItemManager.java [hunk 3]
```diff
@@ -808,14 +808,14 @@ public static String getRealNameByType(EntityType type) {
...
-	DARK_OAK_BUTTON(-1, -1, 6214, "Dark Oak Button", ""),
+	DARK_OAK_BUTTON(143, 0, 6214, "Dark Oak Button", "wooden_button"),
 	DARK_OAK_DOOR(431, 0, 10669, "Dark Oak Door", "DARK_OAK_DOOR_ITEM"),
 	DARK_OAK_FENCE(191, 0, 21767, "Dark Oak Fence", ""),
 	DARK_OAK_FENCE_GATE(186, 0, 10679, "Dark Oak Fence Gate", ""),
 	DARK_OAK_LEAVES(161, 1, 22254, "Dark Oak Leaves", ""),
 	DARK_OAK_LOG(162, 1, 14831, "Dark Oak Log", ""),
 	DARK_OAK_PLANKS(5, 5, 20869, "Dark Oak Wood Plank", "Dark Oak Planks"),
-	DARK_OAK_PRESSURE_PLATE(-1, -1, 31375, "Dark Oak Pressure Plate", ""),
+	DARK_OAK_PRESSURE_PLATE(72, 0, 31375, "Dark Oak Pressure Plate", "Wooden_Presure_Plate"),
```
Comment (montlikadani) [inline src/main/java/com/gamingmesh/jobs/CMILib/ItemManager.java]: Same

File: src/main/java/com/gamingmesh/jobs/CMILib/ItemManager.java [hunk 4]
```diff
@@ -1013,14 +1013,14 @@ public static String getRealNameByType(EntityType type) {
...
-	JUNGLE_BUTTON(-1, -1, 25317, "Jungle Button", ""),
+	JUNGLE_BUTTON(143, 0, 25317, "Jungle Button", "wooden_button"),
 	JUNGLE_DOOR(429, 0, 28163, "Jungle Door", "JUNGLE_DOOR_ITEM"),
 	JUNGLE_FENCE(190, 0, 14358, "Jungle Fence", ""),
 	JUNGLE_FENCE_GATE(185, 0, 21360, "Jungle Fence Gate", ""),
 	JUNGLE_LEAVES(18, 3, 5133, "Jungle Leaves", ""),
 	JUNGLE_LOG(17, 3, 20721, "Jungle Log", ""),
 	JUNGLE_PLANKS(5, 3, 26445, "Jungle Wood Plank", "Jungle Planks"),
-	JUNGLE_PRESSURE_PLATE(-1, -1, 11376, "Jungle Pressure Plate", ""),
+	JUNGLE_PRESSURE_PLATE(72, 0, 11376, "Jungle Pressure Plate", "Wooden_Presure_Plate"),
```
Comment (montlikadani) [inline src/main/java/com/gamingmesh/jobs/CMILib/ItemManager.java]: Same

File: src/main/java/com/gamingmesh/jobs/CMILib/ItemManager.java [hunk 5]
```diff
@@ -1396,14 +1396,14 @@ public static String getRealNameByType(EntityType type) {
...
-	SPRUCE_BUTTON(-1, -1, 23281, "Spruce Button", ""),
+	SPRUCE_BUTTON(143, 0, 23281, "Spruce Button", "wooden_button"),
 	SPRUCE_DOOR(427, 0, 10642, "Spruce Door", "SPRUCE_DOOR_ITEM"),
 	SPRUCE_FENCE(188, 0, 25416, "Spruce Fence", ""),
 	SPRUCE_FENCE_GATE(183, 0, 26423, "Spruce Fence Gate", ""),
 	SPRUCE_LEAVES(18, 1, 20039, "Spruce Leaves", ""),
 	SPRUCE_LOG(17, 1, 9726, "Spruce Log", ""),
 	SPRUCE_PLANKS(5, 1, 14593, "Spruce Wood Plank", "Spruce Planks"),
-	SPRUCE_PRESSURE_PLATE(-1, -1, 15932, "Spruce Pressure Plate", ""),
+	SPRUCE_PRESSURE_PLATE(72, 0, 15932, "Spruce Pressure Plate", "Wooden_Presure_Plate"),
```
Comment (montlikadani) [inline src/main/java/com/gamingmesh/jobs/CMILib/ItemManager.java]: Same

File: src/main/java/com/gamingmesh/jobs/CMILib/ItemManager.java [hunk 6]
```diff
@@ -731,9 +731,9 @@ public static String getRealNameByType(EntityType type) {
 	CARROT(391, 0, 22824, "Carrot", "Carrotitem"),
 	CARROTS(141, 0, 17258, "Carrots", "CARROT"),
 	CARROT_ON_A_STICK(398, 0, 27809, "Carrot on a Stick", "carrotstick"),
-	CARVED_PUMPKIN(-1, -1, 25833, "Carved Pumpkin", ""),
+	CARVED_PUMPKIN(86, 0, 25833, "Carved Pumpkin", "PUMPKIN"),
 	CAULDRON(380, 0, 26531, "Cauldron", "CAULDRON_ITEM"),
-	CAVE_AIR(-1, -1, 17422, "Air", ""),
+	CAVE_AIR(0, 0, 17422, "Air", "AIR"),
```
Comment (montlikadani) [inline src/main/java/com/gamingmesh/jobs/CMILib/ItemManager.java]: CARVED_PUMPKIN and CAVE_AIR ID is invalid.

File: src/main/java/com/gamingmesh/jobs/CMILib/ItemManager.java [hunk 7]
```diff
@@ -755,10 +755,10 @@ public static String getRealNameByType(EntityType type) {
 	CHORUS_PLANT(199, 0, 28243, "Chorus Plant", ""),
 	CLAY(82, 0, 27880, "Clay", "Clay Block"),
 	CLAY_BALL(337, 0, 24603, "Clay Ball"),
-	CLOCK(347, 0, 14980, "Clock", "watch"),
+	CLOCK(347, 0, 14980, "Clock", "WATCH"),
 	COAL(263, 0, 29067, "Coal", ""),
 	COAL_BLOCK(173, 0, 27968, "Block of Coal", ""),
-	COAL_ORE(16, 0, 30965, "Coal Ore", ""),
+	COAL_ORE(16, 0, 30965, "Coal Ore", "COAL_ORE"),
```
Comment (montlikadani) [inline src/main/java/com/gamingmesh/jobs/CMILib/ItemManager.java]: Same, so you do not have to enter the same name of material.

**Comment by montlikadani** [comment]
_This fixes #254_

**Comment by LogGits** [comment]
> 143 ID is same as OAK_BUTTON.

This is intentional. 1.12.2 doesn't have different types of wood buttons.

File: src/main/java/com/gamingmesh/jobs/CMILib/ItemManager.java [hunk 8]
```diff
@@ -768,7 +768,7 @@ public static String getRealNameByType(EntityType type) {
 	COCOA(127, 0, 29709, "Cocoa", ""),
 	COCOA_BEANS(351, 3, 27381, "Coco Beans", "INK_SACK"),
 	COD(349, 0, 24691, "Raw Cod", "RAW_FISH"),
-	COD_BUCKET(-1, -1, 28601, "Bucket of Cod", ""),
+	COD_BUCKET(326, 0, 28601, "Bucket of Cod", "WATER_BUCKET"),
```
Comment (montlikadani) [inline src/main/java/com/gamingmesh/jobs/CMILib/ItemManager.java]: COD_BUCKET is not WATER_BUCKET. And sub-ID is invalid.

File: src/main/java/com/gamingmesh/jobs/CMILib/ItemManager.java [hunk 9]
```diff
@@ -731,9 +731,9 @@ public static String getRealNameByType(EntityType type) {
 	CARROT(391, 0, 22824, "Carrot", "Carrotitem"),
 	CARROTS(141, 0, 17258, "Carrots", "CARROT"),
 	CARROT_ON_A_STICK(398, 0, 27809, "Carrot on a Stick", "carrotstick"),
-	CARVED_PUMPKIN(-1, -1, 25833, "Carved Pumpkin", ""),
+	CARVED_PUMPKIN(91, 0, 25833, "Carved Pumpkin", "JACK_O_LANTERN"),
 	CAULDRON(380, 0, 26531, "Cauldron", "CAULDRON_ITEM"),
-	CAVE_AIR(-1, -1, 17422, "Air", ""),
+	CAVE_AIR(0, 0, 17422, "Air", "AIR"),
```
Comment (montlikadani) [inline src/main/java/com/gamingmesh/jobs/CMILib/ItemManager.java]: CAVE_AIR and CARVED_PUMPKIN sub-ID is invalid. CARVED_PUMPKIN is not JACK_O_LANTERN.

**Comment by LogGits** [comment]
just to clarify. the first two arguments are used for pre 1.13 meaning 1.12.2 down. There is no CAVE_AIR id in that format. 

It is used like this:
`ENUM(id, data, 1.13.1 id, name, 1.12.2 name)`

Air is the closest thing to CAVE_AIR meaning if a user on 1.12.2 uses /give cave_air it will give them air because there is no cave air. Same with carved pumpkins. The closest item to carved pumpkin is a jackolantern. This same thing can be applied to cod_bucket. Its essentially a bucket of water with a fish in it. 1.12.2 doesn't have this and therefore a water bucket is the closest match.

Another major thing. 1.12.2 doesnt have different types of wooden plates/buttons. Meaning this should check for regular buttons and regular pressure plates.

**Comment by LogGits** [comment]
Unless we want to just have it as `-1,-1` and write it up so any -1 = not in 1.12.2. That could also work.

**Comment by LogGits** [comment]
Ok i removed the unnecessary stuff. This should now just resolve carpet issue.

**Comment by montlikadani** [comment]
I accept and merge it because the code currently being edited is correct. The basic code that was edited was incorrect because the sub-ID of item is invalid. If the item has only the ID and the sub-ID is not specified, it will detect that it will give the ID specified.

**Pull request merged**